src/lib/shared/hooks/interfaces/useIHasSelectableItems.ts

```diff
@@ -8,7 +9,7 @@ import { useWebsocketContext } from 'src/lib/utils/useWebsocketContext';
  * @param key key of the device
  * @returns 
  */
-export function useIHasSelectableItems<TState>(key: string): IHasSelectableItemsReturn<TState> | undefined {
+export function useIHasSelectableItems<TState extends DeviceState>(key: string): IHasSelectableItemsReturn<TState> | undefined {
```

Comment:
I'd add a default of `DeviceState` for the type argument here as well. It may not have everything that it should, but it should provide a sensible default in the situations where a type hasn't been provided.